File: app/src/main/java/trust/blockchain/entity/Contract.java
package trust.blockchain.entity;

import android.os.Parcel;
import android.os.Parcelable;
import android.os.Parcelable.Creator;
import trust.blockchain.Slip;

public class Contract implements Parcelable {
    public static final Creator<Contract> CREATOR = new C20801();
    public final Address address;
    public final Slip coin;
    public final String contract;
    public final String name;
    public final String tokenId;
    public final Unit unit;

    /* renamed from: trust.blockchain.entity.Contract$1 */
    static class C20801 implements Creator<Contract> {
        C20801() {
        }

        public Contract createFromParcel(Parcel parcel) {
            return new Contract(parcel);
        }

        public Contract[] newArray(int i) {
            return new Contract[i];
        }
    }

    public Contract(String str, String str2, Unit unit, Slip slip) {
        this(str, "", str2, unit, slip);
    }

    public int describeContents() {
        return 0;
    }

    public String getTokenType() {
        return String.format(this.coin.unit().tokenSymbol, new Object[]{""});
    }

    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(this.coin.name());
        parcel.writeString(this.address.data());
        parcel.writeString(this.contract);
        parcel.writeString(this.name);
        parcel.writeParcelable(this.unit, i);
        parcel.writeString(this.tokenId);
    }

    public Contract(String str, String str2, String str3, Unit unit, Slip slip) {
        this(slip.toAddress(str), str2, str3, unit, slip);
    }

    public Contract(Address address, String str, Unit unit, Slip slip) {
        this(address, "", str, unit, slip);
    }

    public Contract(Address address, String str, String str2, Unit unit, Slip slip) {
        this(address, str, str2, unit, slip, str);
    }

    public Contract(Address address, String str, String str2, Unit unit, Slip slip, String str3) {
        this.address = address;
        this.contract = str;
        this.name = str2;
        this.unit = unit;
        this.coin = slip;
        this.tokenId = str3;
    }

    protected Contract(Parcel parcel) {
        this.coin = Slip.valueOf(parcel.readString());
        this.address = this.coin.toAddress(parcel.readString());
        this.contract = parcel.readString();
        this.name = parcel.readString();
        this.unit = (Unit) parcel.readParcelable(Unit.class.getClassLoader());
        this.tokenId = parcel.readString();
    }
}

File: app/src/main/java/trust/blockchain/entity/Contract.kt
//package trust.blockchain.entity
//
//import android.os.Parcel
//import android.os.Parcelable
//import android.os.Parcelable.Creator
//import trust.blockchain.Slip
//
//class Contract : Parcelable {
//
//    @JvmField val address: Address
//    @JvmField val coin: Slip
//    @JvmField val contract: String
//    @JvmField val name: String
//    @JvmField val tokenId: String
//    @JvmField val unit: Unit
//
//    val tokenType: String
//        get() = String.format(this.coin.unit().tokenSymbol, *arrayOf<Any>(""))
//
//    constructor(str: String, str2: String, unit: Unit, slip: Slip) : this(str, "", str2, unit, slip) {}
//
//    override fun describeContents(): Int {
//        return 0
//    }
//
//    override fun writeToParcel(parcel: Parcel, i: Int) {
//        parcel.writeString(this.coin.name)
//        parcel.writeString(this.address.data())
//        parcel.writeString(this.contract)
//        parcel.writeString(this.name)
//        parcel.writeParcelable(this.unit, i)
//        parcel.writeString(this.tokenId)
//    }
//
//    constructor(str: String, str2: String, str3: String, unit: Unit, slip: Slip) : this(slip.toAddress(str), str2, str3, unit, slip) {}
//
//    constructor(address: Address, str: String, unit: Unit, slip: Slip) : this(address, "", str, unit, slip) {}
//
//    @JvmOverloads
//    constructor(address: Address, str: String, str2: String, unit: Unit, slip: Slip, str3: String = str) {
//        this.address = address
//        this.contract = str
//        this.name = str2
//        this.unit = unit
//        this.coin = slip
//        this.tokenId = str3
//    }
//
//    protected constructor(parcel: Parcel) {
//        this.coin = Slip.valueOf(parcel.readString()!!)
//        this.address = this.coin.toAddress(parcel.readString()!!)
//        this.contract = parcel.readString()!!
//        this.name = parcel.readString()!!
//        this.unit = parcel.readParcelable<Parcelable>(Unit::class.java.classLoader) as Unit
//        this.tokenId = parcel.readString()!!
//    }
//
//    companion object CREATOR : Creator<Contract> {
//        override fun createFromParcel(parcel: Parcel): Contract {
//            return Contract(parcel)
//        }
//
//        override fun newArray(size: Int): Array<Contract?> {
//            return arrayOfNulls(size)
//        }
//    }
//}
